pkg/kubelet/checkpoint/checkpoint.go

```diff
@@ -39,54 +36,57 @@ const (
 	podPrefix = "Pod"
 )
 
-// Manager is the interface used to manage checkpoints
-// which involves writing resources to disk to recover
-// during restart or failure scenarios.
-// https://github.com/kubernetes/community/pull/1241/files
-type Manager interface {
-	// LoadPods will load checkpointed Pods from disk
-	LoadPods() ([]*v1.Pod, error)
-
-	// WritePod will serialize a Pod to disk
-	WritePod(pod *v1.Pod) error
+// Data to be stored as checkpoint
+type Data struct {
+	Pod      *v1.Pod
+	Checksum uint64
+}
 
-	// Deletes the checkpoint of the given pod from disk
-	DeletePod(pod *v1.Pod) error
+// NewPodCheckpoint returns new pod checkpoint
+func NewPodCheckpoint(pod *v1.Pod) checkpointmanager.Checkpoint {
+	return &Data{Pod: pod}
 }
 
-var instance Manager
-var mutex = &sync.Mutex{}
+// MarshalCheckpoint returns marshalled data
+func (cp *Data) MarshalCheckpoint() ([]byte, error) {
+	return json.Marshal(*cp)
+}
 
-// fileCheckPointManager - is a checkpointer that writes contents to disk
-// The type information of the resource objects are encoded in the name
-type fileCheckPointManager struct {
-	path string
+// UnmarshalCheckpoint returns unmarshalled data
+func (cp *Data) UnmarshalCheckpoint(blob []byte) error {
+	err := json.Unmarshal(blob, cp)
+	cksum := cp.Checksum
+	if cksum != cp.GetChecksum() {
+		return errors.ErrCorruptCheckpoint
+	}
+	return err
 }
 
-// NewCheckpointManager will create a Manager that points to the following path
-func NewCheckpointManager(path string) Manager {
-	// NOTE: This is a precaution; current implementation should not run
-	// multiple checkpoint managers.
-	mutex.Lock()
-	defer mutex.Unlock()
-	instance = &fileCheckPointManager{path: path}
-	return instance
+// GetChecksum returns calculated checksum of checkpoint data
+func (cp *Data) GetChecksum() uint64 {
```

Comment:
I dont think this needs to be an exported method.

Reply:
reason for this explained above.